subprojects/core/src/main/java/org/gradle/api/tasks/TaskDestroys.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks;
+
+import org.gradle.api.file.FileCollection;
+
+/**
+ * Represents the files or directories that a {@link org.gradle.api.Task} destroys (removes).
```

Comment:
We should try and keep this open to be about more than files. For example, we _might_ want to use this same mechanism later for tasks that “destroy” running services.

I don't think we need to change anything right now; it's fine to focus on files now. Just something to keep in mind.